### PR TITLE
Topic add pdf

### PR DESCRIPTION
**Pull request by vincecr0ft**
HistiPdfClass now added to histimator/pdfs.py , removing last dependence on probfit

merge_func_code added to histimator/util.py

mask_component_args added to histimator/pdfs.py

### CODE REVIEW

File: histimator/estimator.py [hunk 1]
```diff
@@ -35,28 +46,33 @@ def __init__(self, model, data=None, bins=40, weights=None,
...
+        if constraint <= 0. or isNaN(constraint) : 
+            constraint = 0.
         h_meas = self.h
-        h_pred = np.asarray(
-            [self.pdf(centre[i], self.params) for i in range(bwidth.shape[0])]
-        )
-        h_pred = h_pred*bwidth
         if self.extended:
-            return -st.poisson.logpmf(self.N, h_pred.sum())-poisson.logpmf(h_meas, h_pred).sum()
+            return -st.poisson.logpmf(self.N, h_pred.sum())-poisson.logpmf(h_meas, h_pred).sum() - math.log(constraint)
```
Comment (yhaddad) [inline histimator/estimator.py]: Change `math` into `numpy`

File: histimator/estimator.py [hunk 2]
```diff
@@ -1,9 +1,19 @@
+"""Built-in estimator classes.
+"""
+
+# from . import backend as K
+# if K.backend() == 'tensorflow':
+# import tensorflow as tf
+
+
 import numpy as np
```
Comment (yhaddad) [inline histimator/estimator.py:9]: change this line into `import numpy` to avoid conflicts with the variable `np`

**Review by yhaddad — CHANGES_REQUESTED** [review]
Use `numpy` instead of `math` and avoid conflicts with `np` variable

File: histimator/estimator.py (hunk 2 above)
Comment (vincecr0ft) [reply histimator/estimator.py:9]: ok `math` has been vetoed. 

np will be changed to nuis

**Comment by vincecr0ft** [comment]
hmmmm. The scipy stats dependencies should be removed in another PR. They should be possible to change to be numpy specific, such that it's easier to integrate them with backend options.

**Review by yhaddad — APPROVED** [review]

**Pull request merged**